src/lib/api.js

```diff
@@ -0,0 +1,83 @@
+var defaults = require('lodash.defaults');
+var xhr = require('xhr');
+
+var jar  = require('./jar');
+var log = require('./log');
+var urlParams = require('./url-params');
+
+/**
+ * Helper method that constructs requests to the scratch api.
+ * Custom arguments:
+ *     - useCsrf [boolean] – handles unique csrf token retrieval for POST requests. This prevents
+ *       CSRF forgeries (see: https://www.squarefree.com/securitytips/web-developers.html#CSRF)
+ *
+ * It also takes in other arguments specified in the xhr library spec.
+ */
+
+module.exports = function (opts, callback) {
+    defaults(opts, {
+        host: process.env.API_HOST,
+        headers: {},
+        responseType: 'json',
+        useCsrf: false
+    });
+
+    defaults(opts.headers, {
+        'X-Requested-With': 'XMLHttpRequest'
+    });
+
+    opts.uri = opts.host + opts.uri;
+
+    if (opts.params) {
+        opts.uri = [opts.uri, urlParams(opts.params)]
+            .join(opts.uri.indexOf('?') === -1 ? '?' : '&');
+    }
+
+    if (opts.formData) {
+        opts.body = urlParams(opts.formData);
+        opts.headers['Content-Type'] = 'application/x-www-form-urlencoded';
+    }
+
+    var apiRequest = function (opts) {
+        if (opts.host !== '') {
+            // For IE < 10, we must use XDR for cross-domain requests. XDR does not support
+            // custom headers.
+            defaults(opts, {useXDR: true});
+            delete opts.headers;
+            if (opts.authentication) {
+                var authenticationParams = ['x-token=' + opts.authentication];
+                var parts = opts.uri.split('?');
+                var qs = (parts[1] || '').split('&').concat(authenticationParams).join('&');
+                opts.uri = parts[0] + '?' + qs;
```

Comment:
`useXDR` should use `urlParams`, and needs to be integrated with the `opts.params` above.